pkg/operator/k8sutil/node_test.go

```diff
@@ -389,3 +377,75 @@ func TestGenerateNodeAffinity(t *testing.T) {
 		})
 	}
 }
+
+func TestTopologyLabels(t *testing.T) {
+	additionalTopologyLabels := []string{
+		"rack", "row", "datacenter",
+	}
+	nodeLabels := map[string]string{}
+	topology := ExtractTopologyFromLabels(nodeLabels, additionalTopologyLabels)
+	assert.Equal(t, 0, len(topology))
+
+	// invalid non-namespaced zone and region labels are simply ignored
+	nodeLabels = map[string]string{
+		"region": "badregion",
+		"zone":   "badzone",
+	}
+	topology = ExtractTopologyFromLabels(nodeLabels, additionalTopologyLabels)
+	assert.Equal(t, 0, len(topology))
+
+	// invalid zone and region labels are simply ignored
+	nodeLabels = map[string]string{
+		"topology.rook.io/region": "r1",
+		"topology.rook.io/zone":   "z1",
+	}
+	topology = ExtractTopologyFromLabels(nodeLabels, additionalTopologyLabels)
+	assert.Equal(t, 0, len(topology))
+
+	// load all the expected labels
+	nodeLabels = map[string]string{
+		"topology.kubernetes.io/region": "r1",
```

Comment:
I think it might be good to still test the old beta labels just to make sure they also work.

Reply:
lines 424-431 should cover that case

Reply:
Missed that. Thanks.